templates/pre-job.j2

```diff
@@ -1,11 +1,44 @@
 #!/usr/bin/env bash
 
-GITHUB_SOURCE_REPOSITORY=$(cat "${GITHUB_EVENT_PATH}" | jq -r '.pull_request.head.repo.full_name')
-
-# Request repo-policy-compliance service check.
-curl --noproxy '*' \
-    --fail-with-body \
-    -H 'Authorization: Bearer {{one_time_token}}' \
-    -H 'Content-Type: application/json' \
-    -d "{\"repository_name\": \"${GITHUB_REPOSITORY}\", \"source_repository_name\": \"${GITHUB_SOURCE_REPOSITORY}\", \"target_branch_name\": \"${GITHUB_BASE_REF}\", \"source_branch_name\": \"${GITHUB_HEAD_REF}\", \"commit_sha\": \"${GITHUB_SHA}\"}" \
-    http://{{host_ip}}:8080/check-run
+# Log common env variables.
+logger -s "GITHUB_EVENT_NAME: ${GITHUB_EVENT_NAME}, \
+  GITHUB_REPOSITORY: ${GITHUB_REPOSITORY}, \
+  GITHUB_SHA: ${GITHUB_SHA}"
+
+# Prepare curl auth arguments
+CURL_AUTH_ARGS="--noproxy '*' \
+--fail-with-body \
+-H 'Authorization: Bearer {{one_time_token}}' \
+-H 'Content-Type: application/json'"
+
+# Workflow dispatch - Request repo-policy-compliance service check:
+if [[ "${GITHUB_EVENT_NAME}" == "workflow_dispatch" ]]; then
+
+  logger -s "GITHUB_REF_NAME: ${GITHUB_REF_NAME}"
+
+  curl "${CURL_AUTH_ARGS}" \
```

Comment:
Since this is being changed, what do you think about adding a timeout parameter to curl (-m)?

Reply:
It would be good to add it. Thank you!